routes/v1/posts.js

```diff
@@ -38,7 +38,46 @@ module.exports = function(middleware) {
 			});
 		});
 
-		
+	app.route('/:pid/vote')
+		.post(apiMiddleware.requireUser, function(req, res) {
+			if (!utils.checkRequired(['delta'], req, res)) {
+				return false;
+			}
+			if (typeof req.body.delta !== 'number') {
+				res.status(400).json(errorHandler.generate(
+					400, 'invalid-params',
+					'Required parameters were used incorrectly in this API call, please see the "params" property',
+					['delta']
+				));
+				return false;
+			}
+
+			var payload = {
+				uid: req.user.uid,
+				pid: req.params.pid,
+				delta: req.body.delta
+			}
```

Comment:
No specific need to define `payload` if it is not used in its entirety, looks like later on you just use the properties on their own.